connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -181,6 +184,18 @@ message ExecutePlanResponse {
       string metric_type = 3;
     }
   }
+
+  // This message is only used by dataframe.observe if the observation is Observation.
+  message ObservedMetrics {
+
+    repeated ObservedMetricsObject metrics_objects = 1;
+
+    message ObservedMetricsObject {
+      string name = 1;
+      DataType schema = 2;
+      repeated string values = 3;
```

Comment:
The reason for the previous comment was not necessary to put the schema of the row in every metric, but to figure out if it makes sense to just encode them as literals for now.
```suggestion
      repeated Expression.Literal values = 2;
```

Reply:
Good idea.